src/main/java/de/fraunhofer/aisec/cpg/passes/scopes/ScopeManager.java

```diff
@@ -227,6 +228,8 @@ public void enterScope(Node nodeToScope) {
         newScope = new TryScope(nodeToScope);
       } else if (nodeToScope instanceof NamespaceDeclaration) {
         newScope = new NameScope(nodeToScope, getCurrentNamePrefix(), lang.getNamespaceDelimiter());
+      } else if (nodeToScope instanceof TranslationUnitDeclaration) {
```

Comment:
I had a similar Idea but now we have a global scope, with no node that is created in the scope manager as root node and several global scopes as his children that have a translation-unit (always a file) as its children.

Reply:
Hm yes you are correct and I was actually under the impression that global scopes are limited to a particular TU in C++, but they are not. Top-level declarations of multiple files that are linked together need to be unique, because the global scope is common across all bundled TUs. Meh :(

I will instead just reset the scope to the global scope after each TU entry for now. Reasoning behind this was, that a scope error in one TU A spilled over into the second TU B, making the cpg think, that all declarations of B were in the scope of a class of A.